jdk-tool-plugin/src/main/java/hudson/tools/JDKInstaller.java

```diff
@@ -284,7 +285,9 @@ I tried to locate exactly how InstallShield parses the arguments (and why it use
                 out.println(Messages.JDKInstaller_FailedToInstallJDK(r));
                 // log file is in UTF-16
                 try (InputStreamReader in = new InputStreamReader(fs.read(logFile), "UTF-16")) {
-                    IOUtils.copy(in, new OutputStreamWriter(out, Charset.defaultCharset()));
+                    Computer computer = Computer.currentComputer();
+                    Charset charset = computer != null ? computer.getDefaultCharset() : Charset.defaultCharset();
```

Comment:
I don't think that `computer#getDefaultCharset` can return null here without a previous operation failing. I can add a null check after this line if that's not the case.